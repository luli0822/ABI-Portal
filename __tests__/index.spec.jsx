/* eslint-disable no-unused-vars */
/* eslint-env jest */

import { shallow, mount } from 'enzyme';
import React from 'react';
import Index from '../pages/index';

import {
	mockLang,
	mockPathMap,
	mockLinkedEntries,
	mockLinkedAssets,
	mockSiteEntry,
	mockFieldOrFallback,
	mockLangSwitchLink,
	mockHeaderNav,
	mockSkipNavText,
} from '../contentful/__mocks__/SiteDataContext';

import {
	mockPushCustomLinkEvent,
	mockUsePageViewEvent,
} from '../contentful/__mocks__/analyticsContext';

jest.mock('react', () => {
	const ActualReact = require.requireActual('react');
	return {
		...ActualReact,
		useContext: () => ({
			pathMap: mockPathMap,
			fieldOrFallback: mockFieldOrFallback,
			linkedEntries: mockLinkedEntries,
			linkedAssets: mockLinkedAssets,
			siteEntry: mockSiteEntry,
			langSwitchLink: mockLangSwitchLink,
			headerNav: mockHeaderNav,
			skipNavText: mockSkipNavText,
			lang: mockLang,
			pushCustomLinkEvent: mockPushCustomLinkEvent,
			usePageViewEvent: mockUsePageViewEvent,
		}),
	};
});

let wrapper;

beforeEach(() => {
	wrapper = shallow(<Index pageId="site-home-page" />);
});

describe('Loop page', () => {
	it('renders correctly', () => {
		// https://github.com/facebook/jest/issues/7802
		// using .debug() instead of adding dependency on enzyme-to-json for now
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
