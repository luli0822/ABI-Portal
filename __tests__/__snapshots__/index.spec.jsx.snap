// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loop page renders correctly 1`] = `
"<Fragment>
  <Meta pageId=\\"site-home-page\\" siteDomain=\\"//www.domain.ca\\" />
  <Header navigationEntries={{...}} skipNavText=\\"Skip to main content\\" langSwitchLink={{...}} logo={[undefined]} pushCustomLinkEvent={[Function: mockConstructor]} />
  <main id=\\"main\\" />
  <Footer navigationEntries={{...}} />
</Fragment>"
`;
